### PR TITLE
fix/V1-142 - Time for test and test results for client course

### PR DESCRIPTION
**Pull request by zhuk-vezunchikz**

### CODE REVIEW

File: backend/src/controllers/tests/getTestResults.ts [hunk 1]
```diff
@@ -32,7 +32,19 @@ const getTestResults = async (
     }
 
     const correctAnswers = await getTrueAnswersProvider(testId);
+
     const userWrongAnswers = checkTestResults(answers, correctAnswers.questions);
+
+    const testResultWithAnswers = correctAnswers.questions.map((correctAnswer) => {
+      const result = userWrongAnswers.find((item) => correctAnswer.qN === item.qN);
```
Comment (k-krasilnikova) [inline backend/src/controllers/tests/getTestResults.ts:39]: What is `item` here?

File: backend/src/controllers/tests/getTestResults.ts [hunk 2]
```diff
@@ -32,7 +32,19 @@ const getTestResults = async (
...
+
     const userWrongAnswers = checkTestResults(answers, correctAnswers.questions);
+
+    const testResultWithAnswers = correctAnswers.questions.map((correctAnswer) => {
+      const result = userWrongAnswers.find((item) => correctAnswer.qN === item.qN);
+      if (result) {
+        return { qN: correctAnswer.qN, aN: false };
+      }
+      return { qN: correctAnswer.qN, aN: true };
+    });
```
Comment (k-krasilnikova) [inline backend/src/controllers/tests/getTestResults.ts:44]: Rewrite with using only one `return` statement per function

File: backend/src/db/models/ClientCourses.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ const clientCourseSchema = new Schema<IClientCourse>({
   course: { type: Schema.Types.ObjectId, ref: 'Courses' },
   status: { type: String },
   withAssessment: { type: Boolean, default: false },
-  testResult: { type: String },
+  testResult: [{ qN: { type: Number }, aN: { type: Boolean } }],
```
Comment (k-krasilnikova) [inline backend/src/db/models/ClientCourses.ts:11]: Is it really correct to replace this field with another value? Do we need to change any logic on FE?

**Review by k-krasilnikova — CHANGES_REQUESTED** [review]

File: backend/src/controllers/tests/getTestResults.ts (hunk 1 above)
Comment (zhuk-vezunchikz) [reply backend/src/controllers/tests/getTestResults.ts]: done

File: backend/src/controllers/tests/getTestResults.ts (hunk 2 above)
Comment (zhuk-vezunchikz) [reply backend/src/controllers/tests/getTestResults.ts]: done

File: backend/src/db/models/ClientCourses.ts (hunk 3 above)
Comment (zhuk-vezunchikz) [reply backend/src/db/models/ClientCourses.ts:11]: discussed on the call

**Review by k-krasilnikova — APPROVED** [review]

**Pull request merged**